hazelcast/src/main/java/com/hazelcast/config/DataConnectionConfig.java

```diff
@@ -181,6 +184,26 @@ public DataConnectionConfig setProperties(Properties properties) {
         return this;
     }
 
+    /**
+     * Checks if this configuration object is in coherent state - has all required properties set.
+     */
+    @PrivateApi
+    public void validate() {
```

Comment:
if we really what to do the validation post-creation, then it would be better to have a dedicated validator util class similar to `TieredStoreConfigValidator`, so we do not pollute the API.

Reply:
Done 864528a727a29a37e9d99fe1353876bef2b806ce and later